pincer/objects/guild/guild.py

```diff
@@ -885,6 +880,361 @@ async def delete(self):
+        Returns an async generator of voice regions.
+
+        Returns
+        -------
+        AsyncGenerator[:class:`~pincer.objects.voice.VoiceRegion`, :data:`None`]
+            An async generator of voice regions.
+        """
+        data = await self._http.get(f"guilds/{self.id}/regions")
+        for voice_region_data in data:
+            yield VoiceRegion.from_dict(construct_client_dict(self._client, voice_region_data))
+
+    async def get_invites(self) -> AsyncGenerator[Invite, None]:
+        """|coro|
+        Returns an async generator of invites for the guild.
+        Requires the ``MANAGE_GUILD`` permission.
+
+        Returns
+        -------
+        AsyncGenerator[:class:`~pincer.objects.invite.Invite`, :data:`None`]
+            An async generator of invites.
+        """
+        data = await self._http.get(f"guilds/{self.id}/invites")
+        for invite_data in data:
+            yield Invite.from_dict(construct_client_dict(self._client, invite_data))
+
+    async def get_integrations(self) -> AsyncGenerator[Integration, None]:
+        """|coro|
+        Returns an async generator of integrations for the guild.
+        Requires the ``MANAGE_GUILD`` permission.
+
+        Returns
+        -------
+        AsyncGenerator[:class:`~pincer.objects.integration.Integration`, :data:`None`]
+            An async generator of integrations.
+        """
+        data = await self._http.get(f"guilds/{self.id}/integrations")
+        for integration_data in data:
+            yield Integration.from_dict(construct_client_dict(self._client, integration_data))
+
+    async def delete_integration(
+        self,
+        integration: Integration,
+        reason: Optional[str] = None
+    ):
+        """|coro|
+        Deletes an integration.
+        Requires the ``MANAGE_GUILD`` permission.
+
+        Parameters
+        ----------
+        integration : :class:`~pincer.objects.integration.Integration`
+            The integration to delete.
+        reason : Optional[:class:`str`]
+            Reason for the deletion |default| :data:`None`
+        """
+        await self._http.delete(
+            f"guilds/{self.id}/integrations/{integration.id}",
+            headers=remove_none({"X-Audit-Log-Reason": reason})
+        )
+
+    async def get_widget_settings(self) -> GuildWidget:
+        """|coro|
+        Returns the guild widget settings.
+        Requires the ``MANAGE_GUILD`` permission.
+
+        Returns
+        -------
+        :class:`~pincer.objects.guild.widget.GuildWidget`
+            The guild widget settings.
+        """
+        return GuildWidget.from_dict(
+            construct_client_dict(
+                self._client,
+                await self._http.get(f"guilds/{self.id}/widget")
+            )
+        )
+
+    async def modify_widget(
+        self,
+        reason: Optional[str] = None,
+        **kwargs
+    ) -> GuildWidget:
+        """|coro|
+        Modifies the guild widget for the guild.
```

Comment:
try to specify more what a guild widget is

Reply:
I disagree to add that. It is unnecessary. If one is using the `modify_widget` function they most probably know what a guild widget is.